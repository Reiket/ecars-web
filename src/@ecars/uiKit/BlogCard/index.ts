import {BlogCardContent} from '@ecars/uiKit/BlogCard/components/BlogCardContent';
import {BlogCardComponent} from '@ecars/uiKit/BlogCard/BlogCardComponent';
import {BlogCardImage} from '@ecars/uiKit/BlogCard/components/BlogCardImage';

export const BlogCard = Object.assign(BlogCardComponent, {
  Content: BlogCardContent,
  Image: BlogCardImage,
});
