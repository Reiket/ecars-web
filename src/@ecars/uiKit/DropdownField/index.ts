import {DropdownFieldComponent} from '@ecars/uiKit/DropdownField/DropdownFieldComponent';
import {DropdownFieldWrapper} from '@ecars/uiKit/DropdownField/components/DropdownFieldWrapper';

export const DropdownField = Object.assign(DropdownFieldComponent, {
  Wrapper: DropdownFieldWrapper,
});
