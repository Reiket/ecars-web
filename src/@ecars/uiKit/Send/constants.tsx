import {Icons, withBlockClass} from 'ecars-web-lib';
import {SectionWrapper} from '@ecars/uiKit/Section/components/SectionWrapper';
import type {ReactNode} from 'react';
import type {ButtonColorType} from 'ecars-web-lib/dist/components/Button/constants';

export const SendWrapperHOC = withBlockClass(SectionWrapper, 'send');

interface SendItemType {
  icon: ReactNode;
  title: string;
  text: string;
  buttonText: string;
  buttonColor: ButtonColorType;
}

export const sendItems: SendItemType[] = [
  {
    icon: <Icons.Account />,
    title: 'Send an inquiry and our managers will offer you the best deals.',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.',
    buttonText: 'Get a quote',
    buttonColor: 'green',
  },
  {
    icon: <Icons.Email />,
    title: 'Do you still have questions? Contact us for answers.',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.',
    buttonText: 'Contact Us',
    buttonColor: 'white',
  },
];
