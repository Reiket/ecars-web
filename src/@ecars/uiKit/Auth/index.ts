import {AuthLayout} from '@ecars/uiKit/Auth/components/AuthLayout';
import {AuthPageWrapperHOC} from '@ecars/uiKit/Auth/constants';

export const AuthPage = Object.assign(AuthLayout, {
  Wrapper: AuthPageWrapperHOC,
});