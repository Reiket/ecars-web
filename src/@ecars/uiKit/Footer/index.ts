import {FooterComponent} from '@ecars/uiKit/Footer/FooterComponent';
import {FooterCopyWrapperHOC, FooterWrapperHOC} from '@ecars/uiKit/Footer/constants';
import {FooterCopy} from '@ecars/uiKit/Footer/components/FooterCopy';
import {FooterCopyList} from '@ecars/uiKit/Footer/components/FooterCopyList';
import {FooterColumn} from '@ecars/uiKit/Footer/components/FooterColumn';
import {FooterLogoBlock} from '@ecars/uiKit/Footer/components/FooterLogoBlock';
import {FooterLinkList} from '@ecars/uiKit/Footer/components/FooterLinkList';

export const Footer = Object.assign(FooterComponent, {
  Wrapper: FooterWrapperHOC,
  Column: FooterColumn,
  LogoBlock: FooterLogoBlock,
  LinkList: FooterLinkList,
});

export const Copy = Object.assign(FooterCopy, {
  Wrapper: FooterCopyWrapperHOC,
  List: FooterCopyList,
});
