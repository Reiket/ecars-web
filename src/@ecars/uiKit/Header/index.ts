import {HeaderComponent} from '@ecars/uiKit/Header/HeaderComponent';
import {HeaderNavbarWrapperHOC, HeaderWrapperHOC} from '@ecars/uiKit/Header/constants';
import {HeaderNavbar} from '@ecars/uiKit/Header/components/HeaderNavbar';
import {HeaderUtils} from '@ecars/uiKit/Header/components/HeaderUtils';
import {HeaderBurger} from '@ecars/uiKit/Header/components/HeaderBurger';
import {HeaderNavbarList} from '@ecars/uiKit/Header/components/HeaderNavbarList';

export const Header = Object.assign(HeaderComponent, {
  Wrapper: HeaderWrapperHOC,
  Utils: HeaderUtils,
  Burger: HeaderBurger,
});

export const Navbar = Object.assign(HeaderNavbar, {
  Wrapper: HeaderNavbarWrapperHOC,
  List: HeaderNavbarList,
});
