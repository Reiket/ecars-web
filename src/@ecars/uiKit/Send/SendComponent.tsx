import type {FC} from 'react';
import {Send} from '@ecars/uiKit/Send';
import {sendItems} from '@ecars/uiKit/Send/constants';

export const SendComponent: FC = () => {
  return (
    <Send.Wrapper name="send">
      <div className="send__body">
        {sendItems.map((item) => (
          <Send.Column
            key={item.title}
            {...item}
          />
        ))}
      </div>
    </Send.Wrapper>
  );
};
