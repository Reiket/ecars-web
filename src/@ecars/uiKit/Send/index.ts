import {SendWrapperHOC} from '@ecars/uiKit/Send/constants';
import {SendComponent} from '@ecars/uiKit/Send/SendComponent';
import {SendColumn} from '@ecars/uiKit/Send/components/SendColumn';

export const Send = Object.assign(SendComponent, {
  Wrapper: SendWrapperHOC,
  Column: SendColumn,
});
