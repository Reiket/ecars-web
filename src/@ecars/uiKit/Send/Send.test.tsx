import {render} from '@testing-library/react';
import {Send} from '@ecars/uiKit/Send';

describe('Send Component', () => {
  test('render component correctly', () => {
    const {container} = render(<Send />);
    expect(container).toMatchSnapshot();
    expect(container).toBeInTheDocument();
  });
});
