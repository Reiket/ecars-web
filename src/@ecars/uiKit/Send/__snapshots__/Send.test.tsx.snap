// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Send Component > render component correctly 1`] = `
<div>
  <div
    class="send"
  >
    <section
      class="send__section section"
    >
      <div
        class="send__container container container--lg"
      >
        <div
          class="send__body"
        >
          <div
            class="send__col"
          >
            <div
              class="send__icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12.1597 16C10.1243 16 8.29182 16.8687 7.01276 18.2556C8.38039 19.3474 10.114 20 12 20C13.9695 20 15.7727 19.2883 17.1666 18.1081C15.8956 16.8074 14.1219 16 12.1597 16ZM12 4C7.58172 4 4 7.58172 4 12C4 13.8106 4.6015 15.4807 5.61557 16.8214C7.25639 15.0841 9.58144 14 12.1597 14C14.6441 14 16.8933 15.0066 18.5218 16.6342C19.4526 15.3267 20 13.7273 20 12C20 7.58172 16.4183 4 12 4ZM12 5C14.2091 5 16 6.79086 16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5ZM12 7C10.8954 7 10 7.89543 10 9C10 10.1046 10.8954 11 12 11C13.1046 11 14 10.1046 14 9C14 7.89543 13.1046 7 12 7Z"
                />
              </svg>
            </div>
            <h2
              class="send__title block-title"
            >
              Send an inquiry and our managers will offer you the best deals.
            </h2>
            <p
              class="send__text"
            >
              Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
            </p>
            <button
              class="button button--big button--green"
              data-testid="buttonTestId"
              type="button"
            >
              Get a quote
            </button>
          </div>
          <div
            class="send__col"
          >
            <div
              class="send__icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </div>
            <h2
              class="send__title block-title"
            >
              Do you still have questions? Contact us for answers.
            </h2>
            <p
              class="send__text"
            >
              Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.
            </p>
            <button
              class="button button--big button--white"
              data-testid="buttonTestId"
              type="button"
            >
              Contact Us
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
